packages/ui/src/EmptyState.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { FC } from 'react'
+import cn from 'classnames'
+import { ChevronRight } from 'react-feather'
+
+import { ButtonTypeAnchorProps, ButtonProps, Button } from './Button'
+import { Link } from './Link'
+import { Icon, IconProps } from './Icon'
+
+import styles from './EmptyState.module.scss'
+
+type EmptyStateSize = 'normal' | 'large'
+
+type EmptyStateActionButton = {
+  direction?: 'vertical'
+  actionOnClick: ButtonProps['onClick']
+  actionHref?: never
+  actionTarget?: never
+  actionRel?: never
+} & {
+  size?: EmptyStateSize
+}
+
+type EmptyStateActionLink = {
+  direction: 'horizontal'
+  actionHref: ButtonTypeAnchorProps['href']
+  actionTarget?: ButtonTypeAnchorProps['target']
+  actionRel?: ButtonTypeAnchorProps['rel']
+  actionOnClick?: never
+} & {
+  size?: never
+}
+
+type EmptyStateAction = {
+  action: string
+} & (EmptyStateActionButton | EmptyStateActionLink)
+
+export type EmptyStateProps = {
+  title: string
+  description?: string
+  icon: IconProps['icon']
+  iconLabel: string
+} & EmptyStateAction
+
+/**
+ * ### Purpose
+ * Sometimes users performs an action, that yields an empty result. Such cases may be first use of a feature, view with no data, user cleared view or an error state.
+ * Help the user understand the situation by providing them with guidance in form of an Empty State.
+ *
+ * ### General Info
+ * - There are 2 variations of Empty State: vertical (available in normal and large size) and horizontal (only one size). The use depends on available space in UI.
+ * - Every empty state must contain title (should be specific and clear).
+ * - Good practice is add an action button to guide a user for the next steps to get out of the empty state.
+ * - Description helps to describe the semantics more thoroughly.
+ * - For horizontal type of empty state, use Link component instead of classic button.
```

Comment:
This specific line is not very useful here. It is a guide for the one who implements this component not for the one who uses the component.
```suggestion
```